src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/NoDeploymentsResourcesRuleTests.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Rules;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests;
+
+[TestClass]
+public class NoDeploymentsResourcesRuleTests : LinterRuleTestsBase
+{
+    private void CompileAndTest(string text, int expectedDiagnosticCount, Options? options = null)
+    {
+        AssertLinterRuleDiagnostics(NoDeploymentsResourcesRule.Code, text, expectedDiagnosticCount, options);
+    }
+
+    [DataRow("""
+param name string
+param specId string
+resource foo 'Microsoft.Resources/deployments@2021-04-01' = {
+  name: name
+  properties: {
+    mode: 'Incremental'
+    templateLink: {
+      uri: specId
+    }
+    parameters: {}
+  }
+}
+""")]
+    [DataRow("""
+param name string
+resource foo 'Microsoft.Resources/deployments@2021-04-01' existing = {
+  name: name
+}
+""")]
+    [DataTestMethod]
+    public void If_UsesBothMetadataPropertyAndConflictingDecorator_ShouldRaise(string text)
```

Comment:
Might want to change the name of this method

Reply:
🤦 fixed